source/val/validate_atomics.cpp

```diff
@@ -55,7 +55,8 @@ spv_result_t AtomicsPass(ValidationState_t& _, const Instruction* inst) {
       if (_.HasCapability(SpvCapabilityKernel) &&
           (opcode == SpvOpAtomicLoad || opcode == SpvOpAtomicExchange ||
            opcode == SpvOpAtomicCompareExchange)) {
-        if (!_.IsFloatScalarType(result_type) &&
+        if (!(_.IsFloatScalarType(result_type) &&
```

Comment:
This code is gated by the Kernel capability. That should never be present in Vulkan.